src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/CompletionHandler.cs

```diff
@@ -153,59 +151,59 @@ internal class CompletionHandler : IRequestHandler<CompletionParams, SumType<Com
             {
                 // Set some context on the CompletionItem so the CompletionResolveHandler can handle it accordingly.
                 result = SetResolveData(result.Value, serverKind);
-                if (IsSimpleImplicitExpression(documentSnapshot, request, serverKind))
+
+                if (serverKind == LanguageServerKind.CSharp)
                 {
-                    result = DoNotPreselect(result.Value);
-                    result = IncludeCSharpKeywords(result.Value, serverKind);
+                    result = MassageCSharpCompletions(request, documentSnapshot, result);
                 }
             }
 
             return result;
         }
 
-        private bool IsSimpleImplicitExpression(LSPDocumentSnapshot documentSnapshot, CompletionParams request, LanguageServerKind serverKind)
+        private SumType<CompletionItem[], CompletionList>? MassageCSharpCompletions(
+            CompletionParams request,
+            LSPDocumentSnapshot documentSnapshot,
+            SumType<CompletionItem[], CompletionList>? result)
```

Comment:
This method depends on this not being `null` (otherwise .Value below would cause all sorts of chaos). I'd recommend making this non-nullable